### DISCUSSION TITLE
Reading files in queue / numering files

### DISCUSSION THREAD

**Discussion by wojtek3** [djuseeq/Ch376msc#48] [Q&A]
Hi!
I want to read mp3 files with long name. How can I access to they? Is it possible to open next file, when current file reading ends? Can I access to files via number? I don't see such a possibility, how can I solve it? 

Cheers!

**Comment by djuseeq** [comment]
Hi!
Long file name is not supported by the library at this moment, maybe in the future. If you list the files, you will get the short names of the files, with that name is possible to open them. Example: TextFile.Mine.txt becomes TEXTFI~1.TXT [(source)](https://en.wikipedia.org/wiki/8.3_filename) . No, you need the file name to access the file. A possible solution is, list the files, take the first file name and open. When you finished the file operation, close it, list the files again and open the second one. I'm using this technic  [here](https://github.com/djuseeq/Ch376msc/blob/master/examples/lcd_menu/lcd_menu.ino)
